### PR TITLE
disappearing messages vs share extension

### PR DESCRIPTION
**Pull request by michaelkirk**
based on https://github.com/WhisperSystems/Signal-iOS/pull/2938 so review that first.

~~Note this is a pretty sparse implementation - we don't expose how long the timer is, or offer a way to disable it in this UI.~~ **edit**: apparently it was *too* sparse. It now includes the disappearing timer duration.

Also note that "blocked" takes precedence over the timer. Rationale being that knowing about disappearing message status isn't very important for someone until you can message them.

timers

### CODE REVIEW

**Comment by michaelkirk** [comment]
I've revamped this to incorporate design feedback. PTAL @charlesmchen 

screen shot 2018-01-22 at 10 19 59 am

File: SignalMessaging/categories/UIColor+OWS.m [hunk 1]
```diff
@@ -27,6 +27,11 @@ + (UIColor *)ows_blackColor
     return [UIColor colorWithRed:8.f / 255.f green:10.f / 255.f blue:0. / 255.f alpha:1.f];
 }
 
++ (UIColor *)ows_blackIconColor
```
Comment (charlesmchen) [inline SignalMessaging/categories/UIColor+OWS.m]: Is `ows_blackIconColor` the best name for something that isn't black?

File: SignalServiceKit/src/Contacts/OWSDisappearingMessagesConfiguration.m [hunk 2]
```diff
@@ -73,55 +73,101 @@ + (NSString *)stringForDurationSeconds:(uint32_t)durationSeconds
...
     } else if (durationSeconds < secondsPerWeek) { // Multiple Days
-        amountFormat = NSLocalizedString(@"TIME_AMOUNT_DAYS",
-            @"{{number of days}} embedded in strings, e.g. 'Alice updated disappearing messages "
-            @"expiration to {{5 days}}'. See other *_TIME_AMOUNT strings");
+        if (useShortFormat) {
+            amountFormat = NSLocalizedString(@"TIME_AMOUNT_DAYS_SHORT_FORMAT",
+                                             @"Label text below navbar button, embeds {{number of days}}. Must be very short, like 1 or 2 characters, The space is intentionally ommitted between the text and the embedded duration so that we get, e.g. '5d' not '5 d'. See other *_TIME_AMOUNT strings");
+        } else {
+            amountFormat = NSLocalizedString(@"TIME_AMOUNT_DAYS",
+                                             @"{{number of days}} embedded in strings, e.g. 'Alice updated disappearing messages "
```
Comment (charlesmchen) [inline SignalServiceKit/src/Contacts/OWSDisappearingMessagesConfiguration.m]: Nice job the comments for the translators.

File: SignalMessaging/categories/UIColor+OWS.m (hunk 1 above)
Comment (michaelkirk) [reply SignalMessaging/categories/UIColor+OWS.m]: "dark white"?

**Pull request merged**